setup_caret.sh

```diff
@@ -10,17 +10,29 @@ function show_usage() {
     echo "    -h or --help: show help message"
     echo "    -c or --no-interactive"
     echo "    -n or --no-package-install"
+    echo "    -d or --ros-distro"
     exit 0
 }
 
+ALLOWED_ROS_DISTRO=("humble" "iron")
+
+function validate_ros_distro() {
+    if [[ ! " ${ALLOWED_ROS_DISTRO[*]} " =~ $1 ]]; then
```

Comment:
This condition allows incomplete words such as "ro". I suggest using more robust condition. I do not have a good idea about how to use `ALLOWED_ROS_DISTRO` variable in this condition.
```suggestion
    if [[ ! "$1" =~ ^(humble|iron)$ ]]; then
```